src/sentry/dynamic_sampling/prioritise_transactions.py

```diff
@@ -0,0 +1,307 @@
+                    Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - timedelta(hours=6)),
+                    Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                    Condition(Column("metric_id"), Op.EQ, metric_id),
+                ],
+                granularity=Granularity(3600),
+                orderby=[
+                    OrderBy(Column("org_id"), Direction.ASC),
+                ],
+            )
+            .set_limit(CHUNK_SIZE + 1)
+            .set_offset(offset)
+        )
+        request = Request(
+            dataset=Dataset.PerformanceMetrics.value, app_id="dynamic_sampling", query=query
+        )
+        data = raw_snql_query(
+            request,
+            referrer=Referrer.DYNAMIC_SAMPLING_COUNTERS_FETCH_PROJECTS_WITH_COUNT_PER_TRANSACTION.value,
+        )["data"]
+        count = len(data)
+        more_results = count > CHUNK_SIZE
+        offset += CHUNK_SIZE
+        if more_results:
+            data = data[:-1]
+        for row in data:
+            last_result.append((row["org_id"], row["num_projects"]))
+
+        first_idx = 0
+        count_projects = 0
+        for idx, (org_id, num_projects) in enumerate(last_result):
+            count_projects += num_projects
+            if idx - first_idx >= max_orgs - 1 or count_projects >= max_projects:
+                # we got to the number of elements desired
+                yield [o for o, _ in last_result[first_idx : idx + 1]]
+                first_idx = idx + 1
+                count_projects = 0
+
+        # keep what is left unused from last_result for the next iteration or final result
+        last_result = last_result[first_idx:]
+        if not more_results:
+            break
+    if len(last_result) > 0:
+        yield [org_id for org_id, _ in last_result]
+
+
+def fetch_transactions_with_total_volumes(
+    org_ids: List[int], large_transactions: bool, max_transactions: int
+) -> Iterator[ProjectTransactions]:
+    """
+    Fetch transactions for all orgs and all projects  with pagination orgs and projects with count per root project
+
+    org_ids: the orgs for which the projects & transactions should be returned
+
+    large_transactions: if True it returns transactions with the largest count
+                        if False it returns transactions with the smallest count
+
+    max_transactions: maximum number of transactions to return
+    """
+    start_time = time.time()
+    offset = 0
+    org_ids = list(org_ids)  # just to be sure it is not some other sequence
+    transaction_string_id = indexer.resolve_shared_org("transaction")
+    transaction_tag = f"tags_raw[{transaction_string_id}]"
+    metric_id = indexer.resolve_shared_org(str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value))
+    current_org_id: Optional[int] = None
+    current_proj_id: Optional[int] = None
+    transaction_counts: List[Tuple[str, int]] = []
+
+    if large_transactions:
```

Comment:
```suggestion
    if large_transactions_first:
```

Reply:
The fact that the large transactions come first is not the point, the point of the parameter is that the function fetches large_transactions ( i.e. the sorting is not relevant, what the function returns is).